contracts/PangolinExchange.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.2;
+
+import "@pangolindex/exchange-contracts/contracts/pangolin-periphery/interfaces/IPangolinRouter.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IAssetsExchange.sol";
+import "./SupportedAssets.sol";
+
+/**
+ * @title PangolinExchange
+ * @dev Contract allows user to invest into an ERC20 token
+ * This implementation uses the Pangolin DEX
+ */
+contract PangolinExchange is Ownable, IAssetsExchange {
+  /* ========= STATE VARIABLES ========= */
+  IPangolinRouter pangolinRouter;
+  SupportedAssets supportedAssets;
+
+  /* ========= CONSTRUCTOR ========= */
+
+  constructor (address _pangolinRouter, SupportedAssets _supportedAssets) {
+    pangolinRouter = IPangolinRouter(_pangolinRouter);
+    supportedAssets = _supportedAssets;
+  }
+
+  /* ========= MODIFIERS ========= */
+
+  modifier RefundRemainder {
+    _;
+    if (address(this).balance > 0) {
+      (bool success,) = msg.sender.call{value : address(this).balance}("");
```

Comment:
How about simply using msg.sender.transfer(address(this).balance) ?

Reply:
This was already resolved by removing this modifier and using a `transfer` method:
https://github.com/ava-loan/avaloan/pull/24/files#diff-f367df2379e51902800c734e91c8e85898f9af9f2ec238b1bf1667424118539aR67

and 

https://github.com/ava-loan/avaloan/pull/24/files#diff-f367df2379e51902800c734e91c8e85898f9af9f2ec238b1bf1667424118539aR94